src/main/java/org/commcare/util/JsonUtils.java

```diff
@@ -0,0 +1,24 @@
+package org.commcare.util;
+
+import org.json.JSONArray;
+
+import javax.annotation.Nullable;
+
+public class JsonUtils {
+
+    /**
+     * Converts a JSON Array to a String Array
+     * @param jsonArray A JSON Array containing string objects that we wish to convert to a String Array
+     * @return A String Array representation for the given jsonArray
+     */
+    public static String[] toArray(@Nullable JSONArray jsonArray) {
+        if (jsonArray != null) {
+            String[] stringArray = new String[jsonArray.length()];
+            for (int i = 0; i < jsonArray.length(); i++) {
+                stringArray[i] = jsonArray.optString(i);
```

Comment:
if the item in the list isn't a String this will add a `null` to the output list.

Reply:
````
 public String optString(int index, String defaultValue) {
        Object object = this.opt(index);
        return JSONObject.NULL.equals(object) ? defaultValue : object
                .toString();
    }
````

where `defaultValue` is empty string. So It should be empty if no such key is there otherwise the object would be converted to string.